src/controllers/user.controller.ts

```diff
@@ -0,0 +1,24 @@
+import { Request, Response } from 'express';
+import { IUserAttached } from '../models/user/user.model';
+import { UserService } from '../services/user.service';
+
+const userService = new UserService();
+
+const getUserByEmail = (email: string): Promise<IUserAttached | null> => {
+  return userService.getUserData(email);
+}
+
+export const loginUserViaGoogle = async (verifiedUser: any, req: Request, res: Response): Promise<void> => {
+
+  if (!verifiedUser) res.status(401).send('User does not exist!');
+
+  console.log(verifiedUser);
+
+  const user = await getUserByEmail(verifiedUser.emails[0].value);
+
+  if (user === null) {
+    const savedUser = await userService.saveUserData({userEmail: verifiedUser.emails[0].value, lastLoggedIn: new Date()});
```

Comment:
Could we use built-in timestamp methods in mongo?

Reply:
This is just a tradeoff - we can, but we should change IUserAttached type to can obtain object with $currentDate mongoDB operator and type checking - few advantages, many disadvantages.